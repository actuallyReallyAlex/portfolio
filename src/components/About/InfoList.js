import React from 'react'
import PropTypes from 'prop-types'
import { Box, Heading, Paragraph, Text } from 'grommet'
import { about } from '../../config.json'

const InfoList = ({ size }) => {
  return (
    <Box
      align="start"
      direction={size === 'small' ? 'column' : 'row'}
      fill="horizontal"
      gap={size === 'small' ? 'medium' : 'xlarge'}
      justify="center"
      margin={{ vertical: 'large' }}
    >
      <Box>
        <Heading level="3">Profile</Heading>
        <Paragraph>{about.profile}</Paragraph>
      </Box>
      <Box>
        <Heading className="uppercase" level="3" margin="none">
          Full Name:
        </Heading>
        <Text margin={{ bottom: 'small' }}>{about.fullName}</Text>
        <Heading className="uppercase" level="3" margin="none">
          Job:
        </Heading>
        <Text margin={{ bottom: 'small' }}>{about.job}</Text>
        <Heading className="uppercase" level="3" margin="none">
          Email:
        </Heading>
        <Text>{about.email}</Text>
      </Box>
    </Box>
  )
}

InfoList.propTypes = {
  size: PropTypes.string.isRequired
}

export default InfoList
