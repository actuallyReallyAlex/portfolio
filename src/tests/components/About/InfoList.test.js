import React from 'react'
import InfoList from '../../../components/About/InfoList'
import { shallow } from 'enzyme'

const defaultProps = {
  size: 'medium'
}

describe('<InfoList />', () => {
  test('Should render the <InfoList />.', () => {
    const component = shallow(<InfoList {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
