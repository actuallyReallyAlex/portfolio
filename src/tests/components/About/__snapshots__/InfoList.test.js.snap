// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoList /> Should render the <InfoList />. 1`] = `ShallowWrapper {}`;
